sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -263,6 +265,10 @@ case class Except(left: SparkPlan, right: SparkPlan) extends BinaryNode {
   protected override def doExecute(): RDD[InternalRow] = {
     left.execute().map(_.copy()).subtract(right.execute().map(_.copy()))
   }
+
+  override def outputsUnsafeRows: Boolean = children.forall(_.outputsUnsafeRows)
```

Comment:
If an operator's children produce both unsafe and safe rows, we will convert everything unsafe rows.
so here should be `children.exists(_.outputsUnsafeRows)`